src/vs/workbench/contrib/webview/browser/webviewIconManager.ts

```diff
@@ -8,14 +8,20 @@ import { memoize } from 'vs/base/common/decorators';
 import { URI } from 'vs/base/common/uri';
 import { ILifecycleService, LifecyclePhase } from 'vs/platform/lifecycle/common/lifecycle';
 import { WebviewIcons } from 'vs/workbench/contrib/webview/browser/webview';
+import { IConfigurationService } from 'vs/platform/configuration/common/configuration';
 
 export class WebviewIconManager {
 
 	private readonly _icons = new Map<string, WebviewIcons>();
 
 	constructor(
 		@ILifecycleService private readonly _lifecycleService: ILifecycleService,
-	) { }
+		@IConfigurationService private readonly _configService: IConfigurationService,
+	) {
+		this._configService.onDidChangeConfiguration(_ => {
+			this.updateStyleSheet(this._lifecycleService);
```

Comment:
This should check if the icon theme has actually changed before updating the style sheet. There's a `affectsConfiguration` method on the event that you should be able to use for that